services/src/main/java/io/grpc/protobuf/service/ProtoReflectionService.java

```diff
@@ -201,44 +214,70 @@ private void initFileDescriptorMaps() {
     }
 
     @Override
+    public void run() {
+      if (request != null) {
+        handleReflectionRequest();
```

Comment:
@ejona86  - you mentioned that it's possible for isReady() to be false by the time run() is called. Does that apply to this service, where 1 request = 1 response? I don't think it does, but if so, do I need to do something different to avoid a null pointer exception if request=null in handleReflectionRequest()?